frontend/src/metabase/common/hooks/use-confirmation.tsx

```diff
@@ -43,26 +43,29 @@ export const useConfirmation = () => {
     />
   ) : null;
 
-  const show = ({
-    title,
-    message,
-    onConfirm,
-    onCancel,
-    confirmButtonText = t`Confirm`,
-    cancelButtonText = t`Cancel`,
-    confirmButtonProps,
-    size,
-  }: ConfirmationState) =>
-    setConfirmationState({
+  const show = useCallback(
+    ({
       title,
       message,
       onConfirm,
       onCancel,
-      confirmButtonText,
-      cancelButtonText,
+      confirmButtonText = t`Confirm`,
+      cancelButtonText = t`Cancel`,
       confirmButtonProps,
       size,
-    });
+    }: ConfirmationState) =>
+      setConfirmationState({
+        title,
+        message,
+        onConfirm,
+        onCancel,
+        confirmButtonText,
+        cancelButtonText,
+        confirmButtonProps,
+        size,
+      }),
+    [setConfirmationState],
```

Comment:
Redundant hook dep

Reply:
I'm not saying the hook is redundant. The dep is.

```suggestion
    [],
```